lte/gateway/c/core/oai/tasks/amf/nas_proc.cpp

```diff
@@ -465,4 +465,42 @@ int amf_nas_proc_authentication_info_answer(
 
   OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
 }
+int amf_handle_s6a_update_location_ans(
+    const s6a_update_location_ans_t* ula_pP) {
+  imsi64_t imsi64                   = INVALID_IMSI64;
+  int rc                            = RETURNerror;
+  amf_context_t* amf_ctxt_p         = NULL;
+  ue_m5gmm_context_s* ue_mm_context = NULL;
+  int amf_cause                     = -1;
+  OAILOG_FUNC_IN(LOG_AMF_APP);
+
+  IMSI_STRING_TO_IMSI64((char*) ula_pP->imsi, &imsi64);
+
+  ue_mm_context = lookup_ue_ctxt_by_imsi(imsi64);
+
+  if (ue_mm_context) {
+    amf_ctxt_p = &ue_mm_context->amf_context;
+  }
+
+  if (!(amf_ctxt_p)) {
+    OAILOG_ERROR(LOG_NAS_AMF, "IMSI is invalid\n");
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, RETURNerror);
+  }
+
+  amf_ue_ngap_id_t amf_ue_ngap_id = ue_mm_context->amf_ue_ngap_id;
+
+  OAILOG_DEBUG(
+      LOG_NAS_AMF,
+      "Received update location Answer from Subscriberdb for"
+      " ue_id = %d\n",
+      amf_ue_ngap_id);
+
+  OAILOG_DEBUG(
+      LOG_NAS_AMF, "Received UL rate %" PRIu64 " and DL rate %" PRIu64 "\n",
+      ula_pP->subscription_data.subscribed_ambr.br_ul,
+      ula_pP->subscription_data.subscribed_ambr.br_dl);
+
+  OAILOG_FUNC_RETURN(LOG_NAS_AMF, RETURNok);
+}
```

Comment:
you are just going to print logs and not save the DNN/APN information?

Reply:
In the ongoing changes for integrating with sessiond, we are only populating the default Ambr details to sessiond fetched from subscriberd like below:
message Ambr {
    uint64 br_ul = 1;
    uint64 br_dl = 2;
 
    enum BitrateUnitsAMBR {
          BPS = 0;
          KBPS = 1;
          }
    // Unit (either bps or Kbps)
    BitrateUnitsAMBR br_unit = 3;
   }
In response to this, sessiond will be sending the below details with reference to "message M5GSMSessionContextAccess" from session_manager.proto

PDU session resource request
-------------------------------------------------------
Item 0: id-PDUSessionAggregateMaximumBitRate
    ProtocolIE-Field
        id: id-PDUSessionAggregateMaximumBitRate (130)
        criticality: reject (0)
        value
            PDUSessionAggregateMaximumBitRate
                pDUSessionAggregateMaximumBitRateDL: 1048576bits/s
                pDUSessionAggregateMaximumBitRateUL: 1048576bits/s
Item 3: id-QosFlowSetupRequestList
    ProtocolIE-Field
        id: id-QosFlowSetupRequestList (136)
        criticality: reject (0)
        value
            QosFlowSetupRequestList: 1 item
                Item 0
                    QosFlowSetupRequestItem
                        qosFlowIdentifier: 5
                        qosFlowLevelQosParameters
                            qosCharacteristics: nonDynamic5QI (0)
                                nonDynamic5QI
                                    fiveQI: 9
                            allocationAndRetentionPriority
                                priorityLevelARP: 1
                                pre-emptionCapability: shall-not-trigger-pre-emption (0)
                                pre-emptionVulnerability: not-pre-emptable (0)


PDU session establishment accept
----------------------------------------------------------
QoS rules - Authorized QoS rules
    Length: 9
    QoS rule 1
        QoS rule identifier: 1
        Length: 6
        001. .... = Rule operation code: Create new QoS rule (1)
        ...1 .... = DQR: The QoS rule is the default QoS rule
        .... 0001 = Number of packet filters: 1
        Packet filter 1
            ..11 .... = Packet filter direction: Bidirectional (3)
            .... 0001 = Packet filter identifier: 1
            Length: 1
            Packet filter component 1
                Packet filter component type: Match-all type (1)
        QoS rule precedence: 255
        0... .... = Spare: 0
        .0.. .... = Spare: 0
        ..00 0101 = Qos flow identifier: 5
  
Please suggest your valuable review inputs here

Reply:
I do not follow your comment here. Are you planning to save the APN/DNN data in this PR (to be used for communication with sessiond and then with RAN) or not?

Reply:
2nd PR is ready to be raised which includes changes to storing QoS, APN/DNN info locally in AMF to be communicated to sessiond/RAN